cpp/arcticdb/storage/s3/detail-inl.hpp

```diff
@@ -43,12 +44,12 @@ namespace s3 {
         template<class It>
         using Range = folly::Range<It>;
 
-        template<class S3ClientType, class KeyBucketizer>
+        template<class KeyBucketizer>
         void do_write_impl(
                 Composite<KeySegmentPair> &&kvs,
                 const std::string &root_folder,
                 const std::string &bucket_name,
-                S3ClientType &s3_client,
+                S3ClientWrapper &s3_client,
```

Comment:
We can pass the wrapper by const-ref can't we? At least for some of these methods

Reply:
I've added `const` only to read operations (read, list, exists), to allow for the `MockS3Client` in the child PR to modify it's contents on write/delete